### PR TITLE
Bump LVGL version to 8.3.5

### LINKED ISSUE #66: Update LVGL to 8.x
We're currently using LVGL 7.10, which is out of date by quite a bit. We need to update to 8.x (ideally latest).

### PR DESCRIPTION
**Pull request by nia-e**
Closes #66 
Remaining things to do:

- None!

### CODE REVIEW

**Comment by nia-e** [comment]
This is almost ready to be merged. All that's left is some mild tweaks to `lvgl-codegen`, removing dead code, finishing up the `Style` impl, and redoing examples. Other than that... I think this is ready for review?

File: lvgl/src/lv_core/style.rs [hunk 1]
```diff
@@ -64,1144 +64,197 @@ impl From<Opacity> for u8 {
...
-    pub fn set_outline_pad(&mut self, state: State, value: i16) {
-        let native_state: u32 = state.get_bits();
-        unsafe {
-            lvgl_sys::_lv_style_set_int(
-                self.raw.as_mut(),
-                (lvgl_sys::LV_STYLE_OUTLINE_PAD as u32
-                    | (native_state << lvgl_sys::LV_STYLE_STATE_POS as u32)) as u16,
-                value,
-            );
+macro_rules! gen_lv_style_color {
```
Comment (nia-e) [inline lvgl/src/lv_core/style.rs:195]: Not super happy with the existence of this macro; I'll probably implement `Into<lv_color_t> for Color` so I can reuse the other macro

File: lvgl/src/lv_core/style.rs [hunk 2]
```diff
@@ -64,1144 +64,197 @@ impl From<Opacity> for u8 {
...
+        const COLOR_FILTER_DSC = lvgl_sys::lv_style_prop_t_LV_STYLE_COLOR_FILTER_DSC as u32;
+        const COLOR_FILTER_OPA = lvgl_sys::lv_style_prop_t_LV_STYLE_COLOR_FILTER_OPA as u32;
+        const ANIM_TIME = lvgl_sys::lv_style_prop_t_LV_STYLE_ANIM_TIME as u32;
+        const ANIM_SPEED = lvgl_sys::lv_style_prop_t_LV_STYLE_ANIM_SPEED as u32;
+        const TRANSITION = lvgl_sys::lv_style_prop_t_LV_STYLE_TRANSITION as u32;
+        const BLEND_MODE = lvgl_sys::lv_style_prop_t_LV_STYLE_BLEND_MODE as u32;
+        const LAYOUT = lvgl_sys::lv_style_prop_t_LV_STYLE_LAYOUT as u32;
+        const BASE_DIR = lvgl_sys::lv_style_prop_t_LV_STYLE_BASE_DIR as u32;
+
+        const PROP_ANY = lvgl_sys::lv_style_prop_t_LV_STYLE_PROP_ANY as u32;
```
Comment (nia-e) [inline lvgl/src/lv_core/style.rs:177]: Need to flesh out which constants were missed in the fork I copied this from and/or which were added/removed post-8.1

File: lvgl/src/lv_core/style.rs (hunk 1 above)
Comment (nia-e) [reply lvgl/src/lv_core/style.rs]: Fixed!

**Comment by nia-e** [comment]
Having some trouble getting LVGL to not segfault randomly 😓 is this something you've encountered before @rafaelcaricio? Also, it seems like enabling `lvgl_alloc` leads to an OOM if I don't increase the memory pool. I'm not entirely sure what's going on...

**Comment by nia-e** [comment]
Done! :D I'll fix the old Gauge example (now Meter) later as that seems relatively low priority. All examples build and run as before, all tests pass. 🍾

File: lvgl/src/lv_core/style.rs (hunk 2 above)
Comment (nia-e) [reply lvgl/src/lv_core/style.rs:162]: Doesn't seem like anything relevant is missing, leaving as-is

**Comment by rafaelcaricio** [comment]
@nia-e That is awesome! I will give a look to this PR later today. ;)

**Review by rafaelcaricio — APPROVED** [review]

**Comment by rafaelcaricio** [comment]
@nia-e  Feel free to merge whenever you want. Thank you! 💯

**Pull request merged**